Batteries/Util/Cache.lean

```diff
@@ -64,8 +64,9 @@ def Cache.get [Monad m] [MonadEnv m] [MonadLog m] [MonadOptions m] [MonadLiftT B
       -- TODO: add customization option
       let options := maxHeartbeats.set options <|
         options.get? maxHeartbeats.name |>.getD 1000000
+      let initHeartbeats ← IO.getNumHeartbeats
       let res ← EIO.asTask <|
-        init {} |>.run' {} { options, fileName, fileMap } |>.run' { env }
+        init {} |>.run' {} { options, fileName, fileMap, initHeartbeats } |>.run' { env }
```

Comment:
I see that `Lean.Core.wrapAsync` exists, and it does a bit more heartbeats arithmetic. `Lean.Core.CoreM.toIO` looks relevant.

Reply:
`wrapAsync` indeed looks like the right tool, as otherwise the hearbeat counts on one thread are meaninglessly transferred to another thread.